samples/webapp-spring-boot-oauth2-resource-server/src/main/java/com/vmware/idm/samples/oauth2/VMwareValidateTokenServices.java

```diff
@@ -110,12 +111,12 @@ private void validateAccessTokenLocally(OAuth2AccessToken accessToken) {
             throw new InvalidTokenException(String.format("Invalid issuer: '%s', expected: '%s'", issuer, expectedIssuer));
         }
 
-        // check iat
+        // check iat. Allow for a time skew between the different parties
         Object issuedAt = map.get(ISSUED_AT_KEY);
         if (issuedAt == null || !(issuedAt instanceof Integer)) {
             throw new InvalidTokenException("Missing or invalid 'iat' key, expecting a valid timestamp value.");
         }
-        if ((Integer) issuedAt > now) {
+        if ((Integer) issuedAt > now + ALLOWED_SKEW_IN_MS) {
```

Comment:
if we allow a time skew, do we also want to check that  exp  <= now -ALLOWED_SKEW_IN_MS?
If we use refresh tokens anyways then relaxing the expiry time is probably totally useless, but just wondering.